ouroboros-consensus-shelley-test/test/Test/ThreadNet/RealTPraos.hs

```diff
@@ -78,6 +83,11 @@ instance Arbitrary TestSetup where
       setupD  <- arbitrary
       setupD2 <- arbitrary
 
+      setupInitialNonce <-
```

Comment:
Is there a preferable path for this hash generator?

Reply:
Yeah, I'd go for [`mkNonceFromNumber`](https://github.com/input-output-hk/cardano-ledger-specs/blob/ed7acfc16efba51e10bad3e8fc7f19b043fd8a9a/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/BaseTypes.hs#L201). Maybe we should also sometimes generate `NeutralNonce`.

Reply:
Done, thanks!